client/consensus/slots/src/lib.rs

```diff
@@ -242,6 +250,51 @@ pub trait SimpleSlotWorker<B: BlockT> {
 		Some(proposal)
 	}
 
+	/// Calls `create_inherent_data` and handles errors.
+	async fn create_inherent_data(
+		slot_info: &SlotInfo<B>,
+		logging_target: &str,
+	) -> Option<sp_inherents::InherentData> {
+		let now = Instant::now();
+		let remaining_duration = if now > slot_info.ends_at {
+			Duration::from_millis(0)
+		} else {
+			slot_info.ends_at - now
+		};
+		let inherent_data = match slot_info
+			.create_inherent_data
+			.create_inherent_data()
+			.timeout(remaining_duration)
```

Comment:
The [doc for timeout](https://docs.rs/async-std/latest/async_std/prelude/trait.FutureExt.html#method.timeout) says `Available on unstable only.`. 🤷‍♂️ There is the [timeout function](https://docs.rs/async-std/latest/async_std/future/fn.timeout.html) in the same library.

Reply:
Changed to use delay.